### PR TITLE
WIP PHP7 support

### PR DESCRIPTION
**Pull request by afk11**
Removes support for 5.6 and HHVM (HHVM thinks it's looking at Hack code)

For now the PR's scope is limited to scalar type hints, return types, and declare(static_types=1) everywhere.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -16,13 +16,18 @@
...
+            "name": "Thomas Kerin",
+            "email": "afk11@users.noreply.github.com",
+            "role": "Maintainer"
         }
     ],
     "require": {
-        "php": ">=5.6.0",
+        "php": ">=7.0",
         "ext-gmp": "*",
         "paragonie/random_compat": "^1|^2",
```
Comment (Spomky) [inline composer.json]: Is this dependency still required?

File: composer.json [hunk 2]
```diff
@@ -16,13 +16,18 @@
...
-        "fgrosse/phpasn1": "~1.5"
+        "fgrosse/phpasn1": "v1.5.*"
```
Comment (Spomky) [inline composer.json]: From my understanding `v1.5.*` and `~1.5` are not so different.
IMHO, you should not change it.

WDYT?

File: composer.json (hunk 1 above)
Comment (afk11) [inline composer.json]: Random compat isn't necessary I guess!

File: composer.json (hunk 2 above)
Comment (afk11) [inline composer.json]: Was just being picky, the tilde permits all versions from 1.5 to 2.0. I can change it back, probably no harm leaving it the way it was

Comment (aztech-dev) [inline composer.json]: I think you could use `~1.5.0` but it's a matter of taste really

File: composer.json (hunk 1 above)
Comment (Spomky) [inline composer.json]: `random/compat` provides `random_bytes()` and `random_int()` methods for PHP 5.x.
As PHP 5.x is no longer supported, this dep is useless.

File: composer.json (hunk 2 above)
Comment (Spomky) [inline composer.json]: From my understanding, `~1.5.0` will only allow versions from `>=1.5.0` to `<1.6.0`.
A better solution could be `^1.5` or `^1.5.0`.

See https://blog.madewithlove.be/post/tilde-and-caret-constraints/

Comment (afk11) [inline composer.json]: I suppose you are right, `^1.5` seems a good fit

**Comment by afk11** [comment]
For now I think this PR is good to go with scalar type hints and return types. Anything else can probably follow up in a subsequent PR. ACK's / reviews welcome, be nice to merge this soon!

**Comment by Spomky** [comment]
Looks good to me

**Comment by afk11** [comment]
Rebased

**Pull request merged**